src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MaxLengthAttribute.cs

```diff
@@ -122,16 +122,17 @@ public static bool TryGetCount(object value, out int count)
                 return true;
             }
 
+            Type type = value.GetType();
             PropertyInfo property = null;
             try
             {
-                // On CoreRT, this property may not be enabled for reflection.
-                // It may be possible to eliminate the exception by using direct reflection
-                // (i.e. not via the RuntimeReflectionExtensions or the new split TypeInfo format.
-                property = value.GetType().GetRuntimeProperty("Count");
+                // On CoreRT, this property may not be enabled for reflection.                
+                property = type.GetRuntimeProperty("Count");
             }
-            catch (TypeAccessException)
+            catch (TypeAccessException ex)
             {
+                throw new MissingMemberException(string.Format(CultureInfo.CurrentCulture,
+                    SR.LengthAttribute_CoreRT_TypeAccessException, type), ex);
```

Comment:
@AtsushiKan said in a previous thread, "You should avoid unnecessary use of Reflection when an easy alternative is available (a good example being using Type.Name in exception messages instead of Type.ToString())". So I think this should be `type.Name` instead of just `type` here.

Reply:
The guidance was intended to be parsed in the other direction. `Type.Name` is the dangerous one :-)